xla/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -4264,12 +4264,10 @@ absl::Status MIOpenSupport::DoPoolForward(
                                miopenFloat, pdesc);
       if (cache_hit) {
         // reusing the same buffer
-        workspace = reinterpret_cast<uint8*>(
-            pdesc->workspace->mutable_device_memory()->opaque());
+        workspace = reinterpret_cast<uint8*>(pdesc->workspace->ptr()->opaque());
       } else {
-        wsp_mem = stream->AllocateTemporaryArray<uint8>(workspace_size).value();
-        workspace = reinterpret_cast<uint8*>(
-            wsp_mem->mutable_device_memory()->opaque());
+        wsp_mem = stream->AllocateOwnedArray<uint8>(workspace_size).value();
+        workspace = reinterpret_cast<uint8*>(wsp_mem->ptr()->opaque());
```

Comment:
Hi @klucke 

This commit has broken our build as there's no such method `Stream::AllocateOwnedArray`, there's `StreamExecutor::AllocateOwnedArray`

So it should be something like

```
        wsp_mem = stream->parent()->AllocateOwnedArray<uint8>(workspace_size);
        workspace = reinterpret_cast<uint8*>(wsp_mem.ptr()->opaque());
```

Our fixed PR is on the way. 

Since there is no ROCm CI to check (we're also working on this), you probably wouldn't notice it without running before.

Would be ok just ping me first next time? 

Thanks in advance!